woocommerce/woocommerce-android
# Refresh order list with order status selected filters

Work in progress: #4934


### Description
Part II of the order status filters. In this PR you'll see the changes needed to apply the selected order status filters to the order list. 

In Part I #5039 the UI for selecting order status filters was added. 

One important consideration. In order to be able to merge this PR as well as Part I #5039, changes are protected by feature flag to only display the filters button on debug build. This has some implications to be aware of: 
- Order list tabs have been kept
- All the code to load orders for the different tabs has been kept
- Code for searching orders by order status has also been kept

This means that right now there is a lot of code from `OrderListViewModel` that has to be removed in following PRs, when the feature is ready to be released. This code right now conflicts with the filtering code, so beware the the following "misbehavior" is to be expected at the moment: 
- When filtering order list and then navigating back and forth to order list with filters selected, the filtered list will be lost even though the UI shows filters as selected.  The "processing" orders list will be shown as if we were in the "Processing tab" although we can't see it.
- When navigating back to order list after selecting filters, race conditions may occur between the code that loads the orders for the 2 tabs and the code that loads the filtered order list. This may result into an error. Not to be worried about as code related to loading orders for tabs will be removed and ViewModel will be refactored to load the order list once with the corresponding selected filters.  

### Testing instructions

- Compile debug version
- Open orders list
- Click on filters button
- Select some order status filers
- Go back to order list and see how filters are applied.

### Images/gif

https://user-images.githubusercontent.com/2663464/139015332-ebd2cd60-a999-4520-8375-30bf5683c98e.mp4
